src/python/pants/engine/target.py

```diff
@@ -1612,12 +1616,20 @@ def can_generate(
     def path_globs(self, files_not_found_behavior: FilesNotFoundBehavior) -> PathGlobs:
         if not self.globs:
             return PathGlobs([])
-        error_behavior = files_not_found_behavior.to_glob_match_error_behavior()
+        # Match any if we use default globs, else match all.
```

Comment:
Added comment here, as I had a hard time to grok the meaning of the conditional below.